pkg/alertmanager/distributor_test.go

```diff
@@ -196,6 +198,25 @@ func TestDistributor_DistributeRequest(t *testing.T) {
 			expStatusCode:      http.StatusOK,
 			expectedTotalCalls: 1,
 			route:              "/status",
+		}, {
+			name:               "Read /status should try all alert managers on error",
+			numAM:              3,
+			numHappyAM:         0,
+			replicationFactor:  3,
+			isRead:             true,
+			expStatusCode:      http.StatusInternalServerError,
+			expectedTotalCalls: 3,
+			route:              "/status",
+		}, {
+			name:               "Read /status is sent to 3 AM when 2 are not happy",
+			numAM:              3,
+			numHappyAM:         3,
+			failedReqCount:     2,
```

Comment:
what's the different between failedReqCount and numHappyAM?

Reply:
numHappyAM means number of healthy Alert Managers and failedReqCount is the number of Requests made to these AMs that are failed.